src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+const aveGrade = (points: number[]): number => (
+  points.reduce((sum, point) => sum + point, 0) / points.length);
 
+export type SortOrder = 'asc' | 'desc';
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents: Student[] = [...students];
+
+  copyStudents.sort((studentA: Student, studentB: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+        if (order === 'asc') {
+          return studentA.name.localeCompare(studentB.name);
+        }
```

Comment:
```suggestion
        if (order === 'asc') {
          return studentA[sortBy].localeCompare(studentB[[sortBy]);
        }
```

sortBy contains string value, which can be used as a key inside square brackets, if you rewrite all code like this, you can connect case name && case surname, you can also connect case age && case married, I guess you should also rewrite all code using ternary operator